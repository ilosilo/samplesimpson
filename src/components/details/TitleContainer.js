import * as React from 'react';
import { useState } from 'react';
import {
    View,
    StyleSheet,
    Text,
    TouchableOpacity,
    Modal,
    TextInput
} from 'react-native';
import { saveItem, deleteItem } from '../../config/Storage';

const TitleContainer = (props) => {
    const { item } = props;
    const [title, setTitle] = useState(props.item.job);
    const [changeTitle, setChangeTitle] = useState(false)

    const newItem = {
        id: '',
        name: '',
        avatar: '',
        job: '',
        about: ''
    }
    const setNewTitle = () => {
        newItem.id = item.id;
        newItem.name = item.name;
        newItem.avatar = item.avatar;
        newItem.job = title;
        newItem.about = item.about;
        deleteItem(item).then(() => (saveItem(newItem)));
        item.job = title;
        setChangeTitle(false);
    }
    return (

        <View style={st.titleContainerStyle}>
            <TouchableOpacity
                onPress={() => setChangeTitle(true)}
            >
                {item.name ? <Text style={st.titleNameStyle}>{item.name}</Text> : <Text style={st.titleNameStyle}>Name Unknown</Text>}
                {item.job ? <Text style={st.titleJobStyle}>{item.job}</Text> : <Text style={st.titleJobStyle}>Job Unknown</Text>}
            </TouchableOpacity>

            <Modal
                animationType="slide"
                transparent={true}
                visible={changeTitle}
                onRequestClose={() => {
                    setChangeTitle(!changeTitle);
                }}
            >
                <View style={st.modalStyle}>
                    <Text> Job: </Text>
                    <TextInput style={st.input}

                        value={title}
                        onChangeText={text => setTitle(text)}></TextInput>
                    <View style={{ flexDirection: 'row' }}>
                        <TouchableOpacity
                            onPress={() => { setNewTitle() }}
                            style={st.boxStyle}>
                            <Text> Change Job </Text>
                        </TouchableOpacity>
                        <TouchableOpacity
                            onPress={() => setChangeTitle(false)}
                            style={st.boxStyle}>
                            <Text> Cancel </Text>
                        </TouchableOpacity>
                    </View>

                </View>
            </Modal>
        </View>
    );
};
export default TitleContainer;

const st = StyleSheet.create({
    titleContainerStyle: {
        alignSelf: 'center',
        alignContent: 'center',
        justifyContent: 'center'
    },
    modalStyle: {
        backgroundColor: 'white',
        width: '90%',
        height: 150,
        alignSelf: 'center',
        justifyContent: 'center',
        alignItems: 'center',
        position: 'absolute',
        top: 50,
        borderRadius: 30,
        borderRightWidth: 2,
        borderBottomWidth: 2,
        borderColor: 'gray',
        padding: 30
    },
    boxStyle: {
        borderTopWidth: 2,
        borderLeftWidth: 2,
        borderRightWidth: 2,
        borderColor: 'gray',
        alignItems: 'center',
        marginTop: 30,
        borderRadius: 30,
        padding: 10,
        width: '50%'
    },
    titleNameStyle: {
        fontWeight: 'bold',
        fontSize: 20,
        flexWrap: 'wrap',
        alignSelf: 'center'
    },
    titleJobStyle: {
        fontSize: 15,
        flexWrap: 'wrap',
        alignSelf: 'center'
    },
});
