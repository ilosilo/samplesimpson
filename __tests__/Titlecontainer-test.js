
 import 'react-native';
 import React from 'react';
 import TitleContainer from '../src/components/details/TitleContainer';
 import renderer from 'react-test-renderer';
 
 test('renders correctly', () => {
    const item = {"id":"3","name":"Bart Simpson","avatar":"https://static.wikia.nocookie.net/simpsons/images/6/65/Bart_Simpson.png/revision/latest/scale-to-width-down/250?cb=20190409004756","job":"Student","about":"Bartholomew JoJo \"Bart\" Simpson "}
    
    const tree = renderer.create(<TitleContainer item={item}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });