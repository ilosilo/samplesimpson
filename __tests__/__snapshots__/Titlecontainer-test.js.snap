// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignContent": "center",
      "alignSelf": "center",
      "justifyContent": "center",
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "flexWrap": "wrap",
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    >
      Bart Simpson
    </Text>
    <Text
      style={
        Object {
          "alignSelf": "center",
          "flexWrap": "wrap",
          "fontSize": 15,
        }
      }
    >
      Student
    </Text>
  </View>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "white",
          "borderBottomWidth": 2,
          "borderColor": "gray",
          "borderRadius": 30,
          "borderRightWidth": 2,
          "height": 150,
          "justifyContent": "center",
          "padding": 30,
          "position": "absolute",
          "top": 50,
          "width": "90%",
        }
      }
    >
      <Text>
         Job: 
      </Text>
      <TextInput
        onChangeText={[Function]}
        value="Student"
      />
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "borderColor": "gray",
              "borderLeftWidth": 2,
              "borderRadius": 30,
              "borderRightWidth": 2,
              "borderTopWidth": 2,
              "marginTop": 30,
              "opacity": 1,
              "padding": 10,
              "width": "50%",
            }
          }
        >
          <Text>
             Change Job 
          </Text>
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "borderColor": "gray",
              "borderLeftWidth": 2,
              "borderRadius": 30,
              "borderRightWidth": 2,
              "borderTopWidth": 2,
              "marginTop": 30,
              "opacity": 1,
              "padding": 10,
              "width": "50%",
            }
          }
        >
          <Text>
             Cancel 
          </Text>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
